webapp/src/dogma/features/history/HistoryList.tsx

```diff
@@ -49,21 +43,48 @@ const HistoryList = <Data extends object>({
         cell: (info) => <DateWithTooltip date={info.getValue()} />,
         header: 'Timestamp',
       }),
-      columnHelper.accessor((row: HistoryDto) => row.revision, {
-        cell: (info) => (
-          <NextLink href={`/app/projects/${projectName}/repos/${repoName}/list/${info.row.original.revision}/`}>
-            <Button colorScheme="blue" leftIcon={<FaHistory />} size="sm" onClick={() => handleTabChange(0)}>
-              View Revision {info.row.original.revision}
-            </Button>
-          </NextLink>
-        ),
-        header: 'Actions',
-        enableSorting: false,
-      }),
     ],
     [columnHelper, handleTabChange, projectName, repoName],
   );
-  return <DynamicDataTable columns={columns as ColumnDef<Data>[]} data={data} pagination={true} />;
+  const [{ pageIndex, pageSize }, setPagination] = useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: 10,
+  });
+  const pagination = useMemo(
+    () => ({
+      pageIndex,
+      pageSize,
+    }),
+    [pageIndex, pageSize],
+  );
+  const {
+    data = [],
+    isLoading,
+    error,
+  } = useGetHistoryQuery({
+    projectName,
+    repoName,
+    //  revision starts from -1, for example for pageSize=20
+    //  The first page  /projects/{projectName}/repos/{repoName}/commits/-1?to=20
+    //  The second page /projects/{projectName}/repos/{repoName}/commits/-21?to=20
```

Comment:
Sorry, my previous example was incorrect. 😅 `to` is not the size of commits but a pointer to a specific revision.

Let's say we have two parameters `pageNumber` and `pageSize` and the initial page starts with `1`. We can derive `revision` and `to` parameters with the following formula.
- `revision`: `((pageNumber - 1) * pageSize * -1) - 1`
- `to`: `revision - pageSize + 1`

The final queries are: 
- The first page: `/projects/{projectName}/repos/{repoName}/commits/-1?to=-20`
  - revision = ((1 - 1) * 20 * -1) - 1 = -1
  - to = -1 - 20 + 1 = -20
- The second page: `/projects/{projectName}/repos/{repoName}/commits/-21?to=-40`
  - revision = ((2 - 1) * 20 * -1) - 1 = -21
  - to = -21 - 20 + 1 = -40
- The N page: ...

Reply:
Updated to the following.
```
    //  revision starts from -1, for example for pageSize=20
    //  The first page  /projects/{projectName}/repos/{repoName}/commits/-1?to=-20
    //  The second page /projects/{projectName}/repos/{repoName}/commits/-21?to=-40
    revision: -pageIndex * pageSize - 1,
    to: Math.max(-totalRevision, -(pageIndex + 1) * pageSize),
```